### DISCUSSION TITLE
The Extended address - is it static?

### DISCUSSION THREAD

**Discussion by ksathara** [openthread/openthread#9074] [Q&A] [answered]
Hi,

I have multiple sensor devices and a single leader/router in my use case. I am using the CoAP client/server method to connect and receive data from sensor devices. I was using the client's Ext address to distinguish them when they send data. However I have noted, in very rare occurrences, this extended address changes. I dont know under what circumstances this happen. I was considering using the EUI64 of the clients instead.

1. Can and why the Ext address change in a CoAP client?
2. Can the leader/router get the EUI64 of it's connected children?

Thanks,
Kaushalya

**Comment by jwhui** [comment, marked as answer]
> Can and why the Ext address change in a CoAP client?

Thread devices randomly generate a new Extended Address whenever they are commissioned into a new Thread network. The Thread Device should retain the same Extended Address while it is commissioned to a given Thread network and its non-volatile settings remain intact.

> Can the leader/router get the EUI64 of it's connected children?

Thread Leader / Router do not maintain the Extended Address for all devices. This is by design to limit the amount of state that a given Thread device needs to maintain.

Any Thread device may query the Extended Address from another Thread device using Thread Network Diagnostics. See the [`meshdiag topology`](https://github.com/openthread/openthread/tree/main/src/cli#meshdiag-topology-ip6-addrschildren) CLI command.

**Comment by ksathara** [comment]
Hi Jonathan,

Many thanks for your reply. I can get the Ext addr of connected children 
via child table. Would you be able to comment if I can get the EUI64 of 
the connected children from the leader/router?

Cheers,
Kaushalya


------ Original Message ------
From "Jonathan Hui" ***@***.***>
To "openthread/openthread" ***@***.***>
Cc "Kaushalya Satharasinghe" ***@***.***>; "Author" 
***@***.***>
Date 5/23/2023 2:58:45 AM
Subject Re: [openthread/openthread] The Extended address - is it static? 
(Discussion #9074)

**Reply by jwhui** [reply]
See second half of my response in: https://github.com/openthread/openthread/discussions/9074#discussioncomment-5969905
